File: java/src/main/java/vsdk/toolkit/common/dataStructures/BalancedKDTreeNode.java
package vsdk.toolkit.common.dataStructures;

/**
Node for a balanced kd tree, nodes are placed in arrays
and no loson, hison pointers are necessary
*/

public class BalancedKDTreeNode {
    public Object mData;
    public int mFlags;

    public void copy(KDTreeNode kdNode) {
        mData = kdNode.mData;
        mFlags = kdNode.flags();
    }

    public int discriminator() {
        return mFlags & 0xF;
    }

    public void setDiscriminator(int discr) {
        mFlags = (mFlags & 0xFFF0) | discr;
    }
}

File: typescript/src/main/vsdk/toolkit/common/dataStructures/BalancedKDTreeNode.ts
import { KDTreeNode } from "./KDTreeNode";

export class BalancedKDTreeNode {
  public mData: unknown;
  public mFlags: number;

  public constructor() {
    this.mData = null;
    this.mFlags = 0;
  }

  public copy(kdNode: KDTreeNode): void {
    this.mData = kdNode.mData;
    this.mFlags = kdNode.flags();
  }

  public discriminator(): number {
    return this.mFlags & 0xF;
  }

  public setDiscriminator(discr: number): void {
    this.mFlags = (this.mFlags & 0xFFF0) | discr;
  }
}
